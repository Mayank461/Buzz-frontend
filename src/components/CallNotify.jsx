import React, { useEffect, useState } from 'react';
import { useNavigate } from 'react-router-dom';

const CallNotify = ({ user, socket }) => {
  const [ShowID, setShow] = useState(false);
  const [from, setFrom] = useState(false);
  const navigate = useNavigate();

  useEffect(() => {
    socket.on('CallNotify', (id, from) => {
      setFrom(from);
      setShow(id);
    });
  }, []);

  function accept() {
    setShow(false);
    navigate(`/call?recipient=${from.id}&rpeerid=${from.peerid}`);
  }
  function reject() {
    socket.emit('disconnect-call', from.id, {
      id: user._id,
      picture_url: user.picture_url,
      name: user.firstname + ' ' + user.lastname,
    });
    setShow(false);
  }
  if (ShowID && from)
    return (
      <div className="callNotify d-flex justify-content-around align-items-center">
        <img
          src={from.picture_url || require('../images/blank-profile.png')}
          className="img-fluid"
          alt="pic"
        />
        <div className="col d-flex flex-column">
          <h3 className="mx-2 my-0">{from.name}</h3>
          <div className="d-flex">
            <div className="btn callbtn accept" onClick={accept}>
              <i className="fa fa-phone"></i>
              Accept
            </div>
            <div className="btn callbtn reject" onClick={reject}>
              <i className="fa fa-times"></i>
              Reject
            </div>
          </div>
        </div>
      </div>
    );
};

export default CallNotify;
