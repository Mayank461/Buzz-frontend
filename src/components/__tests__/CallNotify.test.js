import { render, screen } from '@testing-library/react';
import CallNotify from '../CallNotify';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import SocketMock from 'socket.io-mock';

jest.mock('socket.io-client');

describe('Testing connection', () => {
  let socket = new SocketMock();

  it('Should disply ~ New Calling Notification', async () => {
    const user = {
      _id: 1,
      firstname: 'Test',
      lastname: 'User',
      friends: { myFriends: [], mySentRequests: [{}] },
    };
    const data = {
      id: user?._id,
      picture_url: user?.picture_url,
      name: user.firstname + ' ' + user.lastname,
    };

    render(
      <BrowserRouter>
        <CallNotify user={user} socket={socket} />
      </BrowserRouter>
    );
    socket.socketClient.emit('CallNotify', 123, data);

    await new Promise((r) => setTimeout(r, 2000));

    const name = screen.queryByTestId('name-field');
    expect(name).toBeTruthy();

    const accept_btn = screen.queryByTestId('accept-btn');
    expect(accept_btn).toBeTruthy();

    const reject_btn = screen.queryByTestId('reject-btn');
    expect(reject_btn).toBeTruthy();
  });
});
