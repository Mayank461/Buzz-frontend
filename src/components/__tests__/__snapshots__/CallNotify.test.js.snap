// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing connection Should not disply ~ No notification 1`] = `null`;
